tiledb/sm/crypto/test/unit_crypto.cc

```diff
@@ -29,16 +29,45 @@
  * @section DESCRIPTION
  *
  * Tests the `Crypto` class
+ *
+ * @section Test plan
+ *
+ * The random number generator is tested by generating two 64-byte buffers with
+ * cryptographically random data and checking that their content is not the
+ * same. The probability of having the same content is vanishingly small.
+ *
+ * TODO: add documentation for the AES-GCM tests.
+ *
+ * The hash algorithms are tested using official test vectors:
+ *
+ * MD5 test vectors are taken from RFC 1321's section A.5
+ * (https://www.ietf.org/rfc/rfc1321.txt)
+ *
+ * SHA-256 test vectors were taken from SHA256ShortMsg.rsp, from "SHA Test
+ * Vectors for Hashing Bit-Oriented Messages"
+ * (https://csrc.nist.gov/Projects/cryptographic-algorithm-validation-program/Secure-Hashing)
```

Comment:
Exactly what's needed. Citations allow auditing the code against published standards.